aws-s3-transfer-manager/tests/e2e_transfer_test.rs

```diff
@@ -0,0 +1,298 @@
+#![cfg(e2e_test)]
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+// Tests here requires AWS account with pre-configured S3 bucket to run the tests.
+// Refer to https://github.com/awslabs/aws-c-s3/tree/main/tests/test_helper to help set up the S3 in the account
+// Set CRT_S3_TEST_BUCKET_NAME environment variables to the bucket created.
+// By default, it uses s3-tm-rs-test-bucket
+
+use aws_s3_transfer_manager::io::{InputStream, PartData, PartStream, SizeHint, StreamContext};
+use aws_s3_transfer_manager::metrics::unit::ByteUnit;
+use aws_s3_transfer_manager::operation::upload::ChecksumStrategy;
+use aws_sdk_s3::types::ChecksumMode;
+use std::future::Future;
+use std::pin::Pin;
+use std::task::Poll;
+use std::time::Duration;
+use test_common::{drain, global_uuid_str};
+
+use tokio::time::Sleep;
+
+use aws_s3_transfer_manager::types::PartSize;
+
+const PUT_OBJECT_PREFIX: &str = "upload/";
+
+fn get_bucket_names() -> (String, String) {
+    let bucket_name = option_env!("CRT_S3_TEST_BUCKET_NAME")
+        .unwrap_or("aws-c-s3-test-bucket-099565")
+        .to_owned();
+    let express_bucket_name = format!("{}--usw2-az1--x-s3", bucket_name.as_str());
+    (bucket_name, express_bucket_name)
+}
+
+fn generate_key(readable_key: &str) -> String {
+    format!(
+        "{}{}/{}",
+        PUT_OBJECT_PREFIX,
+        global_uuid_str(),
+        readable_key
+    )
+}
+
+async fn test_tm() -> (aws_s3_transfer_manager::Client, aws_sdk_s3::Client) {
+    let tm_config = aws_s3_transfer_manager::from_env()
+        .part_size(PartSize::Target(8 * ByteUnit::Mebibyte.as_bytes_u64()))
+        .load()
+        .await;
+    let client = tm_config.client().clone();
+    let tm = aws_s3_transfer_manager::Client::new(tm_config);
+    (tm, client)
+}
+
+fn create_input_stream(size: usize) -> InputStream {
+    "This is a test"
+        .bytes()
+        .cycle()
+        .take(size)
+        .collect::<Vec<u8>>()
+        .into()
+}
+
+async fn perform_upload(
+    tm: &aws_s3_transfer_manager::Client,
+    bucket_name: &str,
+    key: &str,
+    strategy: Option<ChecksumStrategy>,
+    stream: InputStream,
+) {
+    let mut upload = tm.upload().bucket(bucket_name).key(key).body(stream);
+
+    if let Some(strategy) = strategy {
+        upload = upload.checksum_strategy(strategy);
+    }
+
+    upload.initiate().unwrap().join().await.unwrap();
+}
+
+async fn test_round_trip_helper(file_size: usize, bucket_name: &str, object_key: &str) {
```

Comment:
test_ makes it seem like a test
```suggestion
async fn round_trip_helper(file_size: usize, bucket_name: &str, object_key: &str) {
```

Reply:
Agree